python/lsst/dax/imgserv/api_soda.py

```diff
@@ -124,47 +145,318 @@ def img_capabilities():
+                         _external=True)
+    resp = make_response(render_template("uws_job_result.xml",
+                                         job_id=job_id,
+                                         job_phase=phase,
+                                         job_start_time=start_time,
+                                         job_end_time=end_time,
+                                         job_duration=duration,
+                                         soda_pos=soda_pos,
+                                         job_result_id=job_id,
+                                         job_result=result_url),
+                         HTTPStatus.OK)
+    resp.headers["Content-Type"] = "text/xml"
     return resp
 
 
+@image_soda.route("/async/<job_id>/results/result", methods=["GET"])
+def img_async_job_results_result(job_id: str):
+    """ Get the single result for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml : `str`
+        the job results.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.ready():
+        result = ar.get()  # retrieve path to the image output
+        fn_out = result.get("job_result")
+        resp = send_file(fn_out,
+                         mimetype="image/fits",
+                         as_attachment=True,
+                         attachment_filename=os.path.basename(fn_out))
+        return resp
+    else:
+        return redirect(url_for('api_image_soda.img_async_job',
+                                job_id=job_id,
+                                _external=True))
+
+
+@image_soda.route("/async/<job_id>/parameters", methods=["GET", "POST"])
+def img_async_job_parameters(job_id: str):
+    """Get/Set the parameters for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+    Returns
+    -------
+     xml : `str`
+        the job parameters.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    if ar.state in ["PENDING", "STARTED"]:
+        params=str(ar.args)
+    else:
+        params = "NOT AVAILABLE"
+    if ar.state == "PENDING" and request.method == "POST":
+        # TODO: DM-20852
+        # Should be able to update the job parameters in PENDING state
+        return
+    return _uws_job_response_plain("PARAMS="+params)
+
+
+@image_soda.route("/async/<job_id>/owner", methods=["GET"])
+def img_async_job_owner(job_id: str):
+    """ Get the owner info for the job.
+
+    Parameters
+    ----------
+    job_id : `str`
+
+    Returns
+    -------
+    xml : `str`
+        the job owner.
+    """
+    ar = app_celery.AsyncResult(job_id)
+    user = session["user"] if session.get("user") else "UNKNOWN"
+    if ar.ready():
+        result = ar.get()  # retrieve path to the image output
+        owner = result.get("owner", "UNKNOWN")
+    else:
+        owner = ar.kwargs.get("owner", "UNKNOWN")
+    if owner != user:
+        raise Exception("User in session not matching job owner")
+    return _uws_job_response_plain("OWNER="+owner)
```

Comment:
Why not just return the owner of the job?  I don't think that's exposing any secret information that you have to make sure that it is the same between the person asking for the owner and the person who is the owner.  Why would someone ask if they own all their own jobs?

Plus, if you throw an exception, don't you get a 500?  It seems like if you want to be secure you'd return some kind of 403 forbidden type HTTP code instead.

Reply:
Changed to assert(owner == user) as this shouldn't happen, and return the owner info.